packages/helm-charts/common/templates/_helpers.tpl

```diff
@@ -372,46 +372,51 @@ data:
 {{- end -}}
 
 {{- define "common.node-health-check" -}}
-# fail if any wgets fail
-set -euo pipefail
-RPC_URL=http://localhost:8545
-MAX_LATEST_BLOCK_AGE_SECONDS="{{ .Values.geth.readiness_probe_max_block_age_seconds | default 30 }}"
-MAX_EPOCH_BLOCK_AGE_SECONDS="{{ .Values.geth.readiness_probe_max_block_epoch_age_seconds | default 300 }}"
-EPOCH_SIZE="{{ .Values.genesis.epoch_size | default 17280 }}"
-EPOCH_SIZE_LESS_ONE="$(( $EPOCH_SIZE - 1 ))"
-
-# first check if it's syncing
-SYNCING=$(wget -q --tries=1 --timeout=5 --header "Content-Type: application/json" -O - --post-data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_syncing\",\"params\":[],\"id\":65}" $RPC_URL)
-NOT_SYNCING=$(echo $SYNCING | grep -o '"result":false')
-if [ ! $NOT_SYNCING ]; then
-  echo "Node is syncing: $SYNCING"
-  exit 1
-fi
-
-{{ if .Values.geth.fullnodeCheckBlockAge }}
-# then make sure that the latest block is new
-LATEST_BLOCK_JSON=$(wget -q --tries=1 --timeout=5 --header "Content-Type: application/json" -O - --post-data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_getBlockByNumber\",\"params\":[\"latest\", false],\"id\":67}" $RPC_URL)
-BLOCK_TIMESTAMP_HEX=$(echo $LATEST_BLOCK_JSON | grep -o '"timestamp":"[^"]*' | grep -o '[a-fA-F0-9]*$')
-BLOCK_NUMBER_HEX=$(echo $LATEST_BLOCK_JSON | grep -o '"number":"[^"]*' | grep -o '[a-fA-F0-9]*$')
-BLOCK_TIMESTAMP=$(( 16#$BLOCK_TIMESTAMP_HEX ))
-BLOCK_NUMBER=$(( 16#$BLOCK_NUMBER_HEX ))
-CURRENT_TIMESTAMP=$(date +%s)
-
-# different age allowed for epoch and epoch+1 blocks
-BLOCK_EPOCH_DIFFERENCE=$(( BLOCK_NUMBER % EPOCH_SIZE ))
-case "$BLOCK_EPOCH_DIFFERENCE" in
-  0|$EPOCH_SIZE_LESS_ONE) ALLOWED_AGE="$MAX_EPOCH_BLOCK_AGE_SECONDS" ;;
-  *) ALLOWED_AGE="$MAX_LATEST_BLOCK_AGE_SECONDS" ;;
-esac
-
-# if the most recent block is too old, then indicate the node is not ready
-BLOCK_AGE_SECONDS=$(( $CURRENT_TIMESTAMP - $BLOCK_TIMESTAMP ))
-if [ $BLOCK_AGE_SECONDS -gt $ALLOWED_AGE ]; then
-  echo "Latest block too old. Age: $BLOCK_AGE_SECONDS Block JSON: $LATEST_BLOCK_JSON"
-  exit 1
-fi
-exit 0
-{{- end }}
+function isReady {
+  geth attach << EOF
+
+    // last block max age in seconds
+    var maxAge = 20
+    // minimum peers to consider eth_syncing a good indicator for considering low chances of new block on chain
+    var minPeers = 30
```

Comment:
Can we instead make this a percentage of the configured peers?

With our current settings we have `maxpeers=1150` and `light.maxpeers=1000` that leaves 150 peer slots but with the [default dial ratio, of 3 a node would only dial out to 150/3=50 peers](https://github.com/celo-org/celo-blockchain/blob/33423ce34f62672a0b905d768ff84eddc7d0e7fb/p2p/server.go#L752-L755).

I think we don't want to rely on getting peer counts from inbound connections because that is out of the control of the node, so we should make sure that our `minPeers` is less than the number of peers that a node would dial out to. Which is currently the case but if we changed `maxpeers` to 1050 a node would only try dialling out to 16 nodes and this check would stop working properly.

So I think we could use something like:
```
peers = (maxpeers - lightpeers) * 0.2
if peers > 30 {
  peers = 30
}
if peers < 5 {
  peers = 5 // Below 5 peers the blockchain node will not try to sync.
}
```

Reply:
Thanks for the recommendation @piersy . I have implemented your suggestion to the charts. wdyt?